.github/workflows/docker-publish.yaml

```diff
@@ -0,0 +1,251 @@
+name: DockerPublish
+
+on:
+  workflow_dispatch:
+  push:
```

Comment:
I think you might want to revise when this workflow is triggered. Creating docker images on every push seems like it could be problematic. I'd suggest updating to make docker images on new tags and on pushes just to main. The pushes to main would get the `latest` docker tag and git tags would also be used for docker tags.